db/migrate/20171005113112_move_membershipnumber_to_user.rb

```diff
@@ -0,0 +1,20 @@
+class MoveMembershipnumberToUser < ActiveRecord::Migration[5.1]
+  def change
+    reversible do |direction|
+
+      direction.up do
+        add_column :users, :membership_number, :string, index: :unique
+        add_index :users, :membership_number, unique: true
+        execute "UPDATE users u SET membership_number = ma.membership_number FROM membership_applications ma WHERE u.id = ma.user_id;"
+        remove_column :membership_applications, :membership_number
+      end
+
+      direction.down do
+        add_column :membership_applications, :membership_number, :string, index: :unique
```

Comment:
Please see related comment above. 

When I run `rake db:rollback`, and then regenerate `schema.rb`, the unique index for `membership_number` in this table is not recreated.   I think you need an `add_index` statement here.

Reply:
It can't be rolled back because of the change in logic. Having a unique membership number for the user doesn't mean it's unique for the membership application.

Reply:
Yes, that makes sense - especially since one user can have multiple applications.

This means that this index in our current schema (for `membership_applications`):
```
t.index ["membership_number"], name: "index_membership_applications_on_membership_number", unique: true
```
is _forcing_ different membership numbers for a user with more than one application.  So, in reality, we definitely do _not_ want that unique index back.